app/client/genericState/index.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  createActionTypes,
+  loadActionTypes,
+  modifyActionTypes
+} from "./actions";
+import { sagaFactory } from "./sagas";
+import { combineReducers, ReducersMapObject } from "redux";
+import {
+  byIdReducerFactory,
+  IdListReducerFactory,
+  lifecycleReducerFactory
+} from "./reducers";
+import {
+  EndpointedRegistration,
+  RegistrationMapping,
+  Registration,
+  EndpointedRegistrationMapping
+} from "./types";
+
+export { apiActions } from "./actions";
+export {
+  Registration,
+  LoadRequestAction,
+  CreateRequestAction,
+  ModifyRequestAction
+} from "./types";
+export { Body } from "../api/client/types";
+
+function hasEndpoint(
+  reg: Registration | EndpointedRegistration
+): reg is EndpointedRegistration {
+  return (reg as EndpointedRegistration).endpoint !== undefined;
+}
+
+/**
+ * Creates a set of reducers for each object defined in the registration mapping.
+ * By default creates a map of each instance of an object indexed by the instance's ID. This map is deep-merged
+ * whenever a REST API call triggered by _any_ of the registered objects results in updated data for that object
+ * Additionally if an API endpoint is defined, will also create lifecycle reducers corresponding
+ * to loading, creating and modifying an object, as well as an ordered ID list that is updated when new data is loaded
+ *
+ * @param {RegistrationMapping} registrations - a mapping of each object to be registered
+ * @returns An array of reducers
+ */
+export const createReducers = <R extends RegistrationMapping>(
+  registrations: R
+): ReducersMapObject<R> => {
+  const base: ReducersMapObject = {};
+
+  Object.keys(registrations).map(key => {
+    let reg = { ...registrations[key], name: key };
+
+    let reducers = {
+      byId: byIdReducerFactory(reg)
+    };
+
+    if (hasEndpoint(reg)) {
+      reducers = {
+        ...reducers,
+        ...{
+          loadStatus: lifecycleReducerFactory(loadActionTypes, reg),
+          createStatus: lifecycleReducerFactory(createActionTypes, reg),
+          modifyStatus: lifecycleReducerFactory(modifyActionTypes, reg),
+          IdList: IdListReducerFactory(reg)
```

Comment:
camelCase`idList` is more semantic I think

Reply:
oh yep fair!